app/forms/ConsignmentReferences.scala

```diff
@@ -57,29 +58,29 @@ object ConsignmentReferences {
       model.reference match {
         case Ducr.codeValue => Some((model.reference, Some(model.referenceValue), None))
         case Mucr.codeValue => Some((model.reference, None, Some(model.referenceValue)))
-        case _              => Some(model.reference, None, None)
+        case _              => Some((model.reference, None, None))
     }
 
-  val mapping = Forms
-    .mapping(
-      "reference" -> requiredRadio("consignmentReferences.reference.empty")
-        .verifying("consignmentReferences.reference.error", isContainedIn(allowedReferenceAnswers)),
-      "ducrValue" -> mandatoryIfEqual(
-        "reference",
-        Ducr.codeValue,
-        text()
-          .verifying("consignmentReferences.reference.ducrValue.empty", nonEmpty)
-          .verifying("consignmentReferences.reference.ducrValue.error", isEmpty or validDucrIgnoreCase)
-      ),
-      "mucrValue" -> mandatoryIfEqual(
-        "reference",
-        Mucr.codeValue,
-        text()
-          .verifying("consignmentReferences.reference.mucrValue.empty", nonEmpty)
-          .verifying("consignmentReferences.reference.mucrValue.error", isEmpty or validMucrIgnoreCase)
-      )
-    )(form2Model)(model2Form)
-
-  def form(): Form[ConsignmentReferences] = Form(mapping)
+  def mapping(goodsDirection: JourneyType) =
```

Comment:
Nit: Is the name of this parameter correct? The type is completely different.

Reply:
The two journey types relevant to the ConsignmentReferences screen are ARRIVE & DEPART so I though it was appropriate to say they are directions of travel for the goods. Happy to accept better suggestions though :)

Reply:
I am very caucious with introducing new nomenclature. Especially, since we have similar terminology elsewere (`goodsLocation` for e.g.). I'd simply call it `journeyType` because this is the term used in other places in our services. It confused me already, so it would most certainly confuse someone in the future.